compass/landice/mesh.py

```diff
@@ -407,7 +448,299 @@ def get_dist_to_edge_and_GL(self, thk, topg, x, y, section, window_size=None):
         dist_to_grounding_line[i, j] = dist_to_here_grounding_line.min()
 
     toc = time.time()
-    logger.info('compass.landice.mesh.get_dist_to_edge_and_GL() took {:0.2f} '
+    logger.info('compass.landice.mesh.get_dist_to_edge_and_gl() took {:0.2f} '
                 'seconds'.format(toc - tic))
 
     return dist_to_edge, dist_to_grounding_line
+
+
+def build_cell_width(self, section_name, gridded_dataset,
+                     flood_fill_start=[None, None]):
+    """
+    Determine MPAS mesh cell size based on user-defined density function.
+
+    Parameters
+    ----------
+    section_name : str
```

Comment:
@trhille , I'm curious about the choice of having this as an argument.  It seems in all the test cases using this function, they all use the section `mesh`.  Can you imagine a use case where we wouldn't want to use that section name?  If `build_cell_width` always used the `mesh` section, this argument could be removed. (And same for `build_mali_mesh`.)

Reply:
Yeah, I thought about hard-coding it too. But I can imagine a case in which we want to produce multiple meshes by running a single test case, in which case we might have multiple sections called "mesh_1km", "mesh_2km", "mesh_4km" or similar. So, even though that case would require a bit of extra editing of that particular case's `mesh.py`, I think leaving it like this allows for more flexibility with minimal required changes in the future.

Reply:
Wouldn't it be better, then, to not have the framework take config options but instead take parameters?  That way, the parameters could be computed algorithmically based on a more limited number of config options.  Having a bunch of kind of redundant config sections isn't really consistent with the philosophy of compass.

Reply:
Wouldn't that just be trading redundancy in the config file for redundancy in the code?

Reply:
No, you would have config options that are independent of resolution and a bit of code that can compute the associated resolution-dependent parameters.  This is what we do in the `cosine_bell` test case, which lets a user supply a list of resolutions they would like to test:
https://github.com/MPAS-Dev/compass/blob/main/compass/ocean/tests/global_convergence/cosine_bell/cosine_bell.cfg#L26-L33
https://github.com/MPAS-Dev/compass/blob/main/compass/ocean/tests/global_convergence/cosine_bell/forward.py#L89-L112

The implementation is even a little cleaner in polaris.

Reply:
Same kind of idea in polaris' version of the baroclinic channel:
https://github.com/E3SM-Project/polaris/blob/main/polaris/ocean/tests/baroclinic_channel/baroclinic_channel.cfg#L25-L39
https://github.com/E3SM-Project/polaris/blob/main/polaris/ocean/tests/baroclinic_channel/baroclinic_channel_test_case.py#L58-L64
https://github.com/E3SM-Project/polaris/blob/main/polaris/ocean/tests/baroclinic_channel/forward.py#L138-L154

Reply:
But I don't think you need to change anything in this PR for now, it's just if you find you want to make meshes at a bunch of resolutions from one test case, we should talk about whether there are ways to not hard-code specific resolutions but rather to support any resolution in a reasonable range.  We've found that saves quite a bit of time and adds a lot of flexibility if you decide you want a new resolution (or range of resolutions).

Reply:
After talking with both of you, my preference would be to leave this as it is for now.  The amount of work to modify it seems not worth the trouble for a slight preference.  If/when we implement convergence tests we may want to make other adjustments, so it would be more efficient to wait to make any adjustments until then.

Reply:
Yes, that was my suggestion as well.